torch/csrc/jit/serialization/export.cpp

```diff
@@ -1366,10 +1366,22 @@ void check_onnx_proto(const std::string& proto_string, bool full_check) {
     throw std::runtime_error("Invalid ONNX proto string.");
     return;
   }
+  // baseline check
   onnx::checker::check_model(model);
-
+  onnx::shape_inference::InferShapes(model);
+  // full check including strict shape inference check
+  // Notice that onnx::checker::check_model doesn't
+  // include non-strict model onnx::shape_inference::InferShapes(model).
+  // That's why we need to write another onnx::checker::check_model(model,
+  // full_check) here.
   if (full_check) {
```

Comment:
If full_check is always true, should we remove it from the function?

Reply:
Leaving this for now to see what others think.

Reply:
Why not use `torch.onnx._globals.py::_InternalGlobals.onnx_shape_inference` or similar to control this behavior?

Shape inference is not required by ONNX spec, it is an optional feature. If an invalid ONNX graph is being generated in ONNX converter, we should fix it, regardless of running or not inference shape every time